src/DynamoDbQueryBuilder.php

```diff
@@ -530,6 +530,33 @@ public function findMany($ids, array $columns = [])
         return $collection;
     }
 
+    public function firstOrNew(array $attributes, array $values = [])
+    {
+        if (! is_null($instance = $this->where($attributes)->first())) {
+            return $instance;
+        }
+
+        return $this->model->newInstance($attributes + $values);
+    }
+
+    public function firstOrCreate(array $attributes, array $values = [])
+    {
+        if (! is_null($instance = $this->where($attributes)->first())) {
+            return $instance;
+        }
+
+        return tap($this->model->newInstance($attributes + $values), function ($instance) {
```

Comment:
We don't really need to use `tap` do we? I've learned to not rely too much on Laravel helpers. `tap` isn't available in older version of Laravel.